components/Graphic.jsx

```diff
@@ -234,15 +241,21 @@ var Graphic = React.createClass({
     this.ctx.strokeStyle = "black";
     this.ctx.lineWidth = 1;
     this.ctx.fillStyle = "none";
+    var dpr = window.devicePixelRatio || 1;
+    this.ctx.scale(dpr, dpr);
     this.offset = {x:0, y:0};
     this.colorSeed = 0;
   },
 
   setSize: function(w,h) {
     this.defaultWidth = w;
     this.defaultHeight = h;
-    this.refs.canvas.width = w;
-    this.refs.canvas.height = h;
+    var dpr = window.devicePixelRatio || 1;
```

Comment:
I'd like to reorganize the code here a little bit, just so there's some aliasing for used-more-than-once objects, and operations are grouped:

```
var cvs = this.refs.canvas;
var cstyle = cvs.style;
cstyle.width = this.panelCount * w + "px";
cstyle.height = h + "px";

var dpr = this.getPixelRatio();
cvs.width = this.panelCount * w * dpr;
cvs.height = h * dpr;
this.ctx.scale(dpr, dpr);
```